src/Init/Data/BitVec/Bitblast.lean

```diff
@@ -1230,4 +1230,36 @@ theorem shiftRight_eq_ushiftRightRec (x : BitVec w₁) (y : BitVec w₂) :
   · simp [of_length_zero]
   · simp [ushiftRightRec_eq]
 
+/- ### umod -/
+
+theorem getElem_umod_of_lt {n d : BitVec w} (hi : i < w) (hd : 0#w < d) :
+    (umod n d)[i] = (divRec w {n, d} (DivModState.init w)).r[i] := by
+  rw [← BitVec.umod_eq_divRec] <;> simp [hd]
+
+theorem getElem_umod {n d : BitVec w} (hi : i < w) :
+    (umod n d)[i] = if d = 0#w then n[i] else (divRec w {n, d} (DivModState.init w)).r[i] := by
+  by_cases hd : d = 0#w
+  · simp [hd]
+  · have := (BitVec.not_le (x := d) (y := 0#w)).mp
+    simp only [le_zero_iff] at this
+    rw [← BitVec.umod_eq_divRec (by simp [hd, this])]
+    simp [hd]
+
+theorem getLsbD_umod {n d : BitVec w}:
+    (umod n d).getLsbD i =
+      if d = 0#w then n.getLsbD i
+      else ((divRec w {n, d} (DivModState.init w)).r).getLsbD i := by
+  by_cases hi : i < w
+  · simp only [BitVec.getLsbD_eq_getElem hi, getElem_umod]
```

Comment:
```suggestion
  · simp [BitVec.getLsbD_eq_getElem hi, getElem_umod]
```

Reply:
the `only` is actually necessary here

Reply:
That likely means one of the lemmas in the default simp-set is doing something you don't want. You can check which simp lemmas actually fire without the only with `simp?`, and suppose that suggests `simp only [foo, ...]`, you can prevent `foo` from applying by writing `simp [BitVec.getLsbD_eq_getElem hi, getElem_umod, -foo]` to remove `foo` from the simpset for that call.

Note that this is nice to do if the `simp only` version contains a large number of lemmas that we wouldn't have to repeat with `simp [-foo, ...]`, but that does not seem to be the case here, so I'm fine with either spelling (`simp only` or `simp [-foo, ...]`)